actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -976,18 +974,16 @@ def controller(controller)
         #     namespace :admin, as: "sekret" do
         #       resources :posts
         #     end
-        def namespace(path, options = {}, &block)
-          path = path.to_s
-
-          defaults = {
-            module:         path,
-            as:             options.fetch(:as, path),
-            shallow_path:   options.fetch(:path, path),
-            shallow_prefix: options.fetch(:as, path)
-          }
+        def namespace(name, as: DEFAULT, path: DEFAULT, shallow_path: DEFAULT, shallow_prefix: DEFAULT, **options, &block)
+          name = name.to_s
+          options[:module] ||= name
+          as = name if as == DEFAULT
+          path = name if path == DEFAULT
+          shallow_path = path if shallow_path == DEFAULT
+          shallow_prefix = as if shallow_prefix == DEFAULT
```

Comment:
Are you sure manipulation with `DEFAULT` is required here?

```suggestion
        def namespace(name, as: name, path: name, shallow_path: path, shallow_prefix: as, **options, &block)
          name = name.to_s
          options[:module] ||= name
```

Doubt a little bit about `shallow_prefix` option default.

Reply:
That would assign the non-`to_s`'d value to the default. I'll double check the method docs and test suite to see if it is needed. 👍

Reply:
We need to to_s the value first. I don't like the way this looks either so I'll likely try to deprecate this behaviour so I can use nil values.

Reply:
I came up with the following idea:

```
def namespace(name, __name: name.to_s, as: __name, path: __name, shallow_path: path, shallow_prefix: as, **options, &block)
```

It avoids converting a name to string multiple times but looks somewhat ugly.
I'll let you decide if it is worth it.